src/resource/page/ResourcePage.tsx

```diff
@@ -155,12 +189,25 @@ export default function ResourcePage({ id }: { id: string }): JSX.Element {
 
   return (
     <>
-      <ResourceInfo resource={resource} />
+      <ResourceInfo>
+        <ResourceTitle resource={resource} language={language}>
+          <LanguageSelect
+            id="resource-info-language-select"
+            label="Toimipisteen tietojen kielivalinta"
+            className="resource-info-language-selector"
+            selectedLanguage={language}
+            onSelect={setLanguage}
+            formatter={(selectedLanguage: Language): string =>
+              `Esityskieli: ${selectedLanguage.toUpperCase()}`
+            }
+            theme="dark"
+          />
+        </ResourceTitle>
+        <ResourceAddress resource={resource} language={language} />
+      </ResourceInfo>
       <ResourceDetailsSection id="resource-description" title="Perustiedot">
         <p className="resource-description-text">
-          {hasText(resource?.description?.fi)
-            ? resource?.description?.fi
-            : 'Toimipisteellä ei ole kuvausta.'}
+          {resource?.description[language] || 'Toimipisteellä ei ole kuvausta.'}
```

Comment:
Small improvement suggestion: In case of missing description, we could maybe say it in that language. 🤔

Reply:
I'll change it to, for example"Suomenkielinen toimipisteen kuvaus puuttuu"